### PR TITLE
Help: initiate live chat from Inline Help

### PR DESCRIPTION
**Pull request by lsinger**
Right now, initiating a live chat session / contacting support is somewhat hidden on `/help` → "Contact us". This PR adds the contact form to Inline Help. 

Current status (with forced chat): 

inline-help

Task list:

- [x] add a "Contact us" button to Inline Help
- [x] open the contact form when clicking the "Contact us" button
- [x] close the contact form when clicking the "Contact us" button again or when clicking the search input
- [x] make the contact form initiate an actual chat when submitting the form via "Chat with us", then reset and close the form and Inline Help
- [x] add an A/B test to be able to throttle volume
- [x] make sure we track usage
- [x] look into no selected site when starting from /me (d09ebfd)
- [x] make sure the different support variations work fine when initiated from the inline form
- [x] check sizes on mobile, especially height of the Popover
- [ ] merge Automattic/wp-e2e-tests/pull/1009 when merging this PR

To test: 
- make sure you're in the A/B test variant "inlinecontact" from the "inlineHelpWithContactForm" test (use the bottom right A/B test selection tool in development)
- make sure you can contact support from the inline help popover as shown in the GIF above
- make sure this works across plans and site types
- make sure this works on different screen sizes and devices

### CODE REVIEW

**Comment by jblz** [comment]
@lsinger I just got around to taking a look at this PR. Nice stuff!

What do you think about moving forward with a pared-down version of this form behind an AB Test (with, say, a 5-10% selected size)? That way we wouldn't have to worry much about a huge influx of sessions.

Here are some general thoughts about I had in no particular order:

* Assume the current site (then the primary, if not in a site context) is the pertinent one (it looks like you're already doing something along these lines, cool!).
* Only show the open-ended `textarea` & suppress all the other inputs out for the v1 -- should help us get something out the door ASAP
* Tag the chats somehow that they originated from this interface so we can manually follow up with some selection of them (either directly or in offline analysis)
* If it's easy enough, include the current inline search terms in the chat meta info (so the agent has the benefit of the context)

Once we have a baseline for functionality & session count, then we can start testing various combinations of form functionality to add fidelity to the request for support and to cover more of our customers.

**Comment by lsinger** [comment]
@jblz As discussed on Slack, this PR should be pretty close to shipping functionality-wise because we use the help contact form and slim it down. Adding an A/B test purely for volume control sounds like a neat idea! 👍

File: client/blocks/inline-help/index.jsx [hunk 1]
```diff
@@ -49,6 +54,15 @@ class InlineHelp extends Component {
 		}
 	}
 
+	componentDidUpdate( prevProps, prevState ) {
+		if ( prevState.showContactForm && prevProps.searchQuery !== this.props.searchQuery ) {
+			this.toggleContactForm();
+		}
+		if ( ! prevProps.isHappychatOpen && this.props.isHappychatOpen ) {
+			this.closeInlineHelp();
```
Comment (jblz) [inline client/blocks/inline-help/index.jsx]: Does this mean that when all operators go offline, if someone has the help popover open, it will close?

Comment (lsinger) [reply client/blocks/inline-help/index.jsx]: `isHappychatOpen` is `true` when the Happychat "window" is visible. This line makes sure that we close Inline Help once we've successfully started a Happychat session (also: only if the "window" wasn't open before).

Comment (lsinger) [reply client/blocks/inline-help/index.jsx]: (So, to answer your question — I don't believe so.)

**Comment by mattwondra** [comment]
Great work @lsinger! I've tested every support flow that I can think to test for the inline contact form and they all appear to be working as expected! 🎉 

I did find a few things worth mentioning. Some are more critical than others, but you can decide what's worth fixing before this ships so we can start testing this.

1. When on the /help URL paths, the inline help sidebar icon turns dark blue-gray. In staging the icon itself is white, but in this branch the icon is still dark gray, so it's difficult to see.  
    screen shot 2018-03-06 at 10 52 33 am

2. If you create a brand new account with an unverified email address, a giant warning appears. This could maybe be omitted for the compact view for now, and if we keep this view we can later find a better way to display this warning to customers.
    screen shot 2018-03-06 at 10 59 15 am

3. After submitting a question to Directly you are redirected to /help. It's probably worth putting in the [same protection you did after a Happychat question is submitted](https://github.com/Automattic/wp-calypso/pull/22487/files#diff-bfbe01f0a66b1aa6a6708379328f1d66R141).

4. Explanatory text in Forum and Directly forms is 11px, and feels unreadably small.
    screen shot 2018-03-06 at 11 11 34 am

5. If you choose Spanish language (and possibly other languages), the text for "Contact us" is longer and runs into the arrow icon.
    screen shot 2018-03-06 at 12 07 55 pm

6. When you hover the Sites Dropdown, the inline help box gets a pixel or two taller (I think the shadow being added as a hover state is taking up layout space so the element is getting taller).

7. It feels like there's too much space between the sites dropdown and the question below. Tightening this space up might make it feel more cohesive and comfortable.

8. Since the HelpContact component wasn't meant to be re-used, it renders a `<main role="main">` element. This means that whenever the inline contact form renders there will be two `<main role="main">` on the page. I admittedly know very little about the accessibility implications for this, but it might be worth asking someone who knows better.

File: client/blocks/inline-help/index.jsx (hunk 1 above)
Comment (mattwondra) [reply client/blocks/inline-help/index.jsx]: @lsinger is right — `isHappychatOpen` refers to whether the HC panel is visible or minimized. So this just means any time the Happychat panel opens up, inline help will go away.

There's a few quirks with this approach (if the HC panel is already open and the user opens inline help and fills out the Happychat form, the inline help popover won't go away). But those are rare edge cases that don't affect the primary customer experience, and are fixable with a refactor of the HelpContact form if this experiment is successful and we want to make this perfect for long-term maintenance.

Comment (jblz) [reply client/blocks/inline-help/index.jsx]: ohhhhh!  

I obviously didn't look too deeply & misinterpreted the "open" in `isHappychatOpen` to mean something like "available."

Thanks for clarifying.

**Comment by lsinger** [comment]
@mattwondra thanks so much for that super extensive report — I think I've addressed all of those findings yesterday and today.

**Comment by lsinger** [comment]
Thanks @mattwondra! I'll add some tracking and fix a few issues on small screens, and then ship it.

**Comment by jblz** [comment]
I just gave this another run through and it's working really well.

I noticed one non-blocking thing I'd like to bring up:

When I'm on the `HelpContact` view & close the popover (by clicking or hitting <kbd>esc</kbd>, I'd expect the popover to reset to the results view the next time I open it.

**Comment by lsinger** [comment]
Added tracking and fixed a few issues on mobile screens, especially when held in landscape mode.

**Comment by lsinger** [comment]
Rebased and posted a heads up p2 post p2-pbg9X-cCU.

**Comment by samouri** [comment]
This seems to have made build ~15 KbGz larger: http://iscalypsofastyet.com/push/ea39ce9296700f09945bf0ffbb75f042f9b11631/f84fc89ac65dc8f1d90433bf3d2cb9d2eeef9499.

We should look into async-loading this somehow

**Pull request merged**